### DISCUSSION TITLE
Novice seeking help: error in bpftrace command for detecting the readline function in /bin/bash via kprobe

### DISCUSSION THREAD

**Discussion by try-agaaain** [bpftrace/bpftrace#2575] [Q&A] [answered]
I am new to bpftrace and I wanted to detect the readline function in `/bin/bash` via kprobe, with the help of chatgpt, I wrote the following command:
```console
$ sudo bpftrace -e 'kprobe:/bin/bash:readline { printf("%s readline called in /bin/bash\n", comm); }'
```

But when I execute it, the following error appears in the terminal:
```console
stdin:1:1-26: ERROR: kprobe probe type requires 1 argument
kprobe:/bin/bash:readline { printf("%s readline called in /bin/bash\n", comm); }
```

Can anyone help me to see what the problem is? Any suggestions would be appreciated!

**Comment by viktormalik** [comment, marked as answer]
Hi, for tracing userspace functions, you need to use `uprobe`, i.e. `uprobe:/bin/bash:readline`.

**Reply by try-agaaain** [reply]
Thanks a lot, It's helpful.
